src/functions/delimsizing.js

```diff
@@ -128,12 +128,30 @@ defineFunction({
     handler: (context, args) => {
         const delim = checkDelimiter(args[0], context);
 
-        // \left and \right are caught somewhere in Parser.js, which is
-        // why this data doesn't match what is in buildHTML.
-        return {
-            type: "leftright",
-            value: delim.value,
-        };
+        if (context.funcName === "\\left") {
+            const parser = context.parser;
+            // Parse out the implicit body
+            ++parser.leftrightDepth;
+            const body = parser.parseExpression(false);
+            --parser.leftrightDepth;
+            // Check the next token
+            parser.expect("\\right", false);
+            const right = parser.parseFunction();
+            if (!right) {
+                throw new ParseError('failed to parse function after \\right');
+            }
+            return {
+                type: "leftright",
+                body: body,
+                left: delim.value,
+                right: right.value.value,
+            };
+        } else {
+            return {
+                type: "leftright",
+                value: delim.value,
+            };
```

Comment:
This is a little weird.  We return this object which gets turned into a `ParseNode` which gets returned by `const right = parser.parseFunction();` up above.  I should add a comment here to explain this.

Reply:
Good idea.  I'll add a TODO for that.

Reply:
Looking at a bit more, splitting into `left` and `right` handlers would be tricky b/c the left/right delimiters are matched and the contents in between is assigned an `minner` class.  Ensuring the matching and setting the correct class on the contents would probably be more difficult if they were separate handlers.